pkg/vmsock/conn_linux.go

```diff
@@ -88,13 +88,19 @@ func handleUDP(uConn *net.UDPConn, addr *net.UDPAddr, b []byte) {
 		return
 	}
 	data := make([]byte, UDPMaxBufer)
-	_, err = conn.Read(data)
+	n, err := conn.Read(data)
 	if err != nil {
 		log.Errorf("handleUDP read from vsock host: %v", err)
 		return
 	}
-	// remove the tcp length from the beginning since that is the place for msg id ;b
-	_, err = uConn.WriteToUDP(data[2:], addr)
+	// data[2:n] is to remove the tcp length from the beginning
+	// since that is the place for msg id
+	// IMPORTANT: the write needs to match the length of the data to
+	// the length of the returned slice. This is necessary since some
+	// tools such as busybox nslookup they explicitly check this
+	// https://elixir.bootlin.com/busybox/latest/source/networking/nslookup.c#L640
```

Comment:
We should always pin to a commit (and never `latest`); also, that isn't the canonical repository.
Should probably be https://git.busybox.net/busybox/tree/networking/nslookup.c?h=1_35_0#n640 instead?
It's also unclear to me how that line of source maps to the comment, but okay…

Reply:
I think it is actually the code in [ns_parse.c](https://github.com/bminor/musl/blob/7a43f6fea9081bdd53d8a11cef9e9fab0348c53d/src/network/ns_parse.c#L104) from the musl library that is more relevant, but it doesn't really matter...

Random bonus tip (not applicable here): For repos on Github you can press the `y` key after selecting some lines in a file, and the URL will change to the latest commit that has modified those lines.

Reply:
Sure, I always use `y` if the link is on github to get the canonical form, however, I could not find the source link on github.